src/http/server/response.cr

```diff
@@ -170,6 +170,26 @@ class HTTP::Server
       respond_with_status(HTTP::Status.new(status), message)
     end
 
+    # Sends a redirect to *location*.
+    #
+    # The value of *location* gets encoded with `URI.encode`.
+    #
+    # The *status* determines the HTTP status code which can be
+    # `HTTP::Status::FOUND` (`302`) for a temporary redirect or
+    # `HTTP::Status::MOVED_PERMANENTLY` (`301`) for a permanent redirect.
+    #
+    # The response gets closed.
+    #
+    # Raises `IO::Error` if the response is closed or headers were already
+    # sent.
+    def redirect(location : String | URI | Path, status : HTTP::Status = :found)
+      check_headers
+
+      self.status = status
+      headers["Location"] = URI.encode(location.to_s)
```

Comment:
Another thought of mine is it worth making the header name `location`? Specifically so it'll support HTTP/2 out of the box while still being backwards compatible?

Reply:
I suppose that makes sense, however we're using title case header names everywhere else in `HTTP::Server`, so making this single instance compatible to HTTP/2 wouldn't help anything (it wouldn't anyways because it's not implemented).
So I'd leave it like that. We can update all the header names when support for HTTP/2 arrives.